src/test/ui/structs/80853.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0618]: expected function, found `S`
+  --> $DIR/80853.rs:4:5
+   |
+LL | fn repro_ref(thing: S) {
+   |              ----- is of type `S`
+LL |     thing();
+   |     ^^^^^--
+   |     |
+   |     call expression requires function
+   |
+   = note: local variable `S` is not a function
```

Comment:
This is confusing for me, `S` is not a "local variable", isn't `S` a "unit struct" but it was being used as a generic?